DOCSP-39181: drupal integration page (#1002)
mongodb/docs-ecosystem: source/php-frameworks/drupal.txt

.. _php-drupal-integration:

==========================
Drupal MongoDB Integration
==========================

.. facet::
   :name: genre
   :values: tutorial

.. meta::
   :keywords: php framework, CMS, web app, authentication, Atlas

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

Overview
--------

> In this guide, you can learn about the benefits of using MongoDB Atlas
> as the database for Drupal sites and navigate to a tutorial that
> demonstrates how to set up the MongoDB driver for Drupal.

Reviewer: S: slightly confusing

```suggestion
In this guide, you can learn about the benefits of using MongoDB Atlas
as the database for Drupal sites. The end of the guide links to a tutorial that
demonstrates how to set up the MongoDB driver for Drupal.
```

You could also just lose the second sentence, since a subsequent paragraph mentions the tutorial

`Drupal <https://www.drupal.org/about>`__ is an open-source web content
management system (CMS) written in PHP. Drupal provides simple content
authoring, composability, and robust authentication features.

The :ref:`php-drupal-tutorial` section of this guide links to a tutorial

> which you can follow to set up a Drupal site that connects to MongoDB Atlas.

Reviewer: I: [that/which](https://www.mongodb.com/docs/meta/style-guide/terminology/alphabetical-terms/#std-term-that:~:text=Use%20that%20to%20introduce%20a%20restrictive%20clause%2C)

```suggestion
that you can follow to set up a Drupal site that connects to MongoDB Atlas.
```

Author: this one always gets me

The :ref:`php-drupal-resources` section contains links to resources and
documentation for further learning.

Why Use MongoDB as the Database for Drupal Sites?
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Websites that seek to provide personalized user experiences must contain
robust capabilities for user authentication. Sites that have many
authenticated users might face performance impacts due to the complexity
of retrieving entity data from multiple tables in a relational database.

> MongoDB supports eliminates the need for complicated table joins and

Reviewer: I: assume this should just be 'MongoDB eliminates'?

```suggestion
MongoDB eliminates the need for complicated table joins and
```

Author: yerp

thus increases data retrieval speed, enhancing user experience with
components such as personalized dashboards and dynamic content feeds.

When you use MongoDB Atlas as your site's database, Drupal stores entity
instances as JSON objects that contain all revisions, translations, and
field data. This flexible data structure decreases latency and allows
Drupal to support personalized, user-focused experiences.

MongoDB offers the following features that improve your Drupal sites:

- Horizontal scaling: Distribute loads across multiple
  servers, making your database scalable for large user bases.

- Integrated file storage: Store user files directly in the
  database instead of on the web server, simplifying hosting.

> - Full-Text search: Avoid implementing separate search

Reviewer: ```suggestion
- Full-text search: Avoid implementing separate search
```
